src/invocation/agent.rs

```diff
@@ -204,28 +204,35 @@ where
             .put(cid.clone(), invocation.clone())
             .map_err(ReceiveError::InvocationStoreError)?;
 
-        let proof_payloads: Vec<&delegation::Payload<DID>> = self
+        let proofs = &self
             .delegation_store
             .get_many(&invocation.proofs())
-            .map_err(ReceiveError::DelegationStoreError)?
+            .map_err(ReceiveError::DelegationStoreError)?;
+        let proof_payloads: Vec<&delegation::Payload<DID>> = proofs
             .iter()
-            .map(|d| &d.payload)
-            .collect();
+            .zip(invocation.proofs().iter())
+            .map(|(d, cid)| {
```

Comment:
Perhaps worth making these pairs an option on the store?

Reply:
Maybe? I'm fine either way. Perhaps we just wait for a second case of this code happening, then we abstract?